### PR TITLE
Make serverAddressByClientCIDRs in discovery API optional

### LINKED ISSUE #61868: Discovery API responding with null for required property
**Is this a BUG REPORT or FEATURE REQUEST?**:

/kind bug

**What happened**:
Discovery API `/apis/` and `/apis/<APIGroup>/` respond with `null` for [required](https://github.com/kubernetes/kubernetes/blob/3cb73605aa6679b24a4cefadd41eb2669b06bab9/api/swagger-spec/apis.json#L63-L66) `serverAddressByClientCIDRs` property, which fails the validation in generated python client. (ref: https://github.com/kubernetes-client/python/issues/418)

In `apisHandler` and `apiGroupHandler`, we didn't initialize `serverAddressByClientCIDRs` for the returned `discoveryGroup`:

https://github.com/kubernetes/kubernetes/blob/8b6a4e14ffb6661f987af83e63c59b9cc8632d46/staging/src/k8s.io/kube-aggregator/pkg/apiserver/handler_apis.go#L63

**What you expected to happen**:
`serverAddressByClientCIDRs` in `metav1.APIGroup` should be `optional` if apiserver is not currently using it, otherwise apiserver should initialize an empty slice for it:

https://github.com/kubernetes/kubernetes/blob/3cb73605aa6679b24a4cefadd41eb2669b06bab9/staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/types.go#L795-L802

Note that `serverAddressByClientCIDRs` in `metav1.APIVersions` for legacy api is properly set. 

**How to reproduce it (as minimally and precisely as possible)**:
Bring up a cluster from HEAD and request for `/apis/` or one of the api groups through kubectl proxy or kubectl api-versions, and look for the `null` in response. E.g.:

```
{
  "kind": "APIGroup",
  "apiVersion": "v1",
  "name": "certificates.k8s.io",
  "versions": [
    {
      "groupVersion": "certificates.k8s.io/v1beta1",
      "version": "v1beta1"
    }
  ],
  "preferredVersion": {
    "groupVersion": "certificates.k8s.io/v1beta1",
    "version": "v1beta1"
  },
  "serverAddressByClientCIDRs": null
}
```

/sig api-machinery
/assign

### PR DESCRIPTION
**Pull request by roycaihw**
**What this PR does / why we need it**:
See https://github.com/kubernetes/kubernetes/issues/61868

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #61868

**Special notes for your reviewer**:
WIP: I'm having trouble updating swagger-spec using our update scripts. Thinking about removing swagger-spec from our code base as it has long passed deprecation. Sending this PR now to see the test results. 

**Release note**:

```release-note
Property `serverAddressByClientCIDRs` in `metav1.APIGroup` (discovery API) now become optional instead of required
```

/sig api-machinery

### CODE REVIEW

**Comment by mbohlool** [comment]
We need more of these PRs. There are more properties (mostly lists) that we return null in kubernetes for them and assumption is null is empty. I think you can compile a list of them if you look through python client issues but it would be nice to do a sweep on all arrays/maps in return values in our types and check if they can be null and add optional for them.

Thanks for taking care of this @roycaihw

**Comment by roycaihw** [comment]
Since I saw `update-openapi-spec` and `update-swagger-spec` diverging already in some other PRs, it might worth getting this PR in ignoring `swagger-spec`

/hold cancel
/assign @caesarxuchao 
for approval

File: staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/types.go [hunk 1]
```diff
@@ -799,6 +799,7 @@ type APIGroup struct {
 	// The server returns only those CIDRs that it thinks that the client can match.
 	// For example: the master will return an internal IP CIDR only, if the client reaches the server using an internal IP.
 	// Server looks at X-Forwarded-For header or X-Real-Ip header or request.RemoteAddr (in that order) to get the client IP.
+	// +optional
 	ServerAddressByClientCIDRs []ServerAddressByClientCIDR `json:"serverAddressByClientCIDRs" protobuf:"bytes,4,rep,name=serverAddressByClientCIDRs"`
```
Comment (caesarxuchao) [inline staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/types.go]: I think the json tag needs the `omitempty`.

**Comment by roycaihw** [comment]
> I think the json tag needs the `omitempty`.

@caesarxuchao That's a great point. Updated. 

Also it seems that `update-swagger-spec` is triggered by `omitempty`.

**Pull request merged**